EKF/estimator_interface.cpp

```diff
@@ -248,19 +248,30 @@ void EstimatorInterface::setBaroData(uint64_t time_usec, float data)
 		}
 	}
 
+	// downsample to highest possible sensor rate
+	// by baro data by taking the average of incoming sample
+	_baro_sample_count++;
+	_baro_alt_sum += baro_alt_meter;
+
 	// limit data rate to prevent data being lost
 	if ((time_usec - _time_last_baro) > _min_obs_interval_us) {
 
+		const float baro_alt_avg = _baro_alt_sum / (float)_baro_sample_count;
+
 		baroSample baro_sample_new;
-		baro_sample_new.hgt = data;
-		baro_sample_new.time_us = time_usec - _params.baro_delay_ms * 1000;
+		baro_sample_new.hgt = compensateBaroForDynamicPressure(baro_alt_avg);
 
+		// Use the time in the middle of the downsampling interval for the sample
+		baro_sample_new.time_us = _time_last_baro + (time_usec - _time_last_baro) / 2;
```

Comment:
Not end of integration period?

Reply:
I don't know. The averaging on the Firmware side took the average. Up for discussion

Reply:
The effective sampling time when samples are combined using an arithmetic means is the average of the individual time stamps.